Modules/Filtering/FFT/src/itkVnlFFTImageFilterFactory.cxx

```diff
@@ -32,53 +32,45 @@
 
 namespace itk
 {
-// Undocumented API used to register during static initialization.
-// DO NOT CALL DIRECTLY.
-void ITKFFT_EXPORT
-     VnlComplexToComplex1DFFTFactoryRegister__Private()
+VnlFFTImageFilterFactory::VnlFFTImageFilterFactory()
 {
   ObjectFactoryBase::RegisterInternalFactoryOnce<FFTImageFilterFactory<VnlComplexToComplex1DFFTImageFilter>>();
-}
-
-void ITKFFT_EXPORT
-     VnlComplexToComplexFFTFactoryRegister__Private()
-{
   ObjectFactoryBase::RegisterInternalFactoryOnce<FFTImageFilterFactory<VnlComplexToComplexFFTImageFilter>>();
-}
-
-void ITKFFT_EXPORT
-     VnlForward1DFFTFactoryRegister__Private()
-{
   ObjectFactoryBase::RegisterInternalFactoryOnce<FFTImageFilterFactory<VnlForward1DFFTImageFilter>>();
-}
-
-void ITKFFT_EXPORT
-     VnlForwardFFTFactoryRegister__Private()
-{
   ObjectFactoryBase::RegisterInternalFactoryOnce<FFTImageFilterFactory<VnlForwardFFTImageFilter>>();
-}
-
-void ITKFFT_EXPORT
-     VnlHalfHermitianToRealInverseFFTFactoryRegister__Private()
-{
   ObjectFactoryBase::RegisterInternalFactoryOnce<FFTImageFilterFactory<VnlHalfHermitianToRealInverseFFTImageFilter>>();
+  ObjectFactoryBase::RegisterInternalFactoryOnce<FFTImageFilterFactory<VnlInverse1DFFTImageFilter>>();
+  ObjectFactoryBase::RegisterInternalFactoryOnce<FFTImageFilterFactory<VnlInverseFFTImageFilter>>();
+  ObjectFactoryBase::RegisterInternalFactoryOnce<FFTImageFilterFactory<VnlRealToHalfHermitianForwardFFTImageFilter>>();
 }
 
-void ITKFFT_EXPORT
-     VnlInverse1DFFTFactoryRegister__Private()
+VnlFFTImageFilterFactory::~VnlFFTImageFilterFactory() = default;
```

Comment:
Could this be moved to the header file?

Reply:
destructor should be in compiled files for libraries when possible. The compiler can take advantage that there is only one destructor for a class and associate virtual method tables and runtime type information with where the descriptor is instantiated and compiled making the information unique among all the compilation units. While many systems can resolve this duplication information it is a best practice to have it be unique as is can obvious some problems that can a rise such as issue with dynamic casts.

Reply:
Thanks for the explanation @blowekamp , I wasn't aware of the compiler optimization benefits and that's good to keep in mind. However, having a destructor defined as `default` in the src file here seems to be inconsistent with the rest of ITK and seems to me to reduce header readability.

For instance, see `itkImageBase.h`:
[https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkImageBase.h#L790](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkImageBase.h#L790)

If we want to destructors to source files that's fine, but it would need to be more clearly enumerated or standardized.

Reply:
ImageBase is a templated class and does not have a compiled source for a library (.cxx), it only has a template header file (.hxx).

Reply:
I see the advantages of both limited compilation and ease of readability for a default implementation in the declaration. Without a clear convention in the toolkit and an entry in the Style Guide, I will leave it as-is.